src/path.js

```diff
@@ -16,6 +16,285 @@ function Path() {
+        const token = pathData.charAt(i);
+        const lastBuffer = buffer[buffer.length - 1];
+        if (number.indexOf(token) > -1) {
+            buffer[buffer.length - 1] += token;
+        } else if (sign.indexOf(token) > -1) {
+            if (!command.type && !this.commands.length) {
+                command.type = 'L';
+            }
+
+            if (token === '-') {
+                if (!command.type || lastBuffer.indexOf('-') > 0) {
+                    isUnexpected = true;
+                } else if (lastBuffer.length) {
+                    buffer.push('-');
+                } else {
+                    buffer[buffer.length - 1] = token;
+                }
+            } else {
+                if (!command.type || lastBuffer.length > 0) {
+                    isUnexpected = true;
+                } else {
+                    continue;
+                }
+            }
+        } else if (supportedCommands.indexOf(token) > -1) {
+            if (command.type) {
+                applyCommand.apply(this);
+                command = { type: token };
+            } else {
+                command.type = token;
+            }
+        } else if (unsupportedCommands.indexOf(token) > -1) {
+            // TODO: try to interpolate commands not directly supported?
+            throw new Error('Unsupported path command: ' + token + '. Currently supported commands are ' + supportedCommands.split('').join(', ') + '.');
+        } else if (' ,\t\n\r\f\v'.indexOf(token) > -1) {
+            buffer.push('');
+        } else if (token === '.') {
+            if (!command.type || lastBuffer.indexOf(token) > -1) {
+                isUnexpected = true;
+            } else {
+                buffer[buffer.length - 1] += token;
+            }
+        } else {
+            isUnexpected = true;
+        }
+
+        if (isUnexpected) {
+            throw new Error('Unexpected character: ' + token + ' at offset ' + i);
+        }
+    }
+    applyCommand.apply(this);
+
+    if (options.optimize) {
+        this.commands = optimizeCommands(this.commands);
+    }
+
+    let flipY = options.flipY;
+    if (flipY === true) {
+        const boundingBox = this.getBoundingBox();
+        flipY = boundingBox.y1 + boundingBox.y2;
+    }
+    // apply x/y offset, flipping and scaling
+    for (const i in this.commands) {
+        const cmd = this.commands[i];
+        for (const prop in cmd) {
+            if (['x', 'x1', 'x2'].includes(prop)) {
+                this.commands[i][prop] = options.x + cmd[prop] * options.scale;
+            } else if (['y', 'y1', 'y2'].includes(prop)) {
+                this.commands[i][prop] = options.y + (flipY !== false ? flipY - cmd[prop] : cmd[prop]) * options.scale;
+            }
+        }
+    }
+
+    return this;
+};
+
+/**
+ * Generates a new Path() from an SVG path element or path notation
+ * @param  {string|SVGPathElement}
+ * @param  {object}
+ */
+Path.fromSVG = function(path, options) {
```

Comment:
Don't offer much gain, not sure about it usefulness

Reply:
Well, we use such shorthands in other places of the codebase too, and I liked the shorter notation without having to create an empty Path first.